src/CoreWCF.RabbitMQ/tests/IntegrationTests.cs

```diff
@@ -1,63 +1,247 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
+using System.Net;
 using System.Threading.Tasks;
 using Contracts;
 using CoreWCF.Channels;
+using CoreWCF.Channels.Configuration;
 using CoreWCF.Configuration;
 using CoreWCF.Queue.Common.Configuration;
 using CoreWCF.RabbitMQ.Tests.Helpers;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.DependencyInjection;
+using RabbitMQ.Client;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace CoreWCF.RabbitMQ.Tests
 {
-    public class IntegrationTests
+    public class IntegrationTests : IClassFixture<ConnectionSettingsFixture>
     {
         private readonly ITestOutputHelper _output;
-        public const string QueueName = "wcfQueue";
+        public static ConnectionSettingsFixture ConnectionSettingsFixture;
 
-        public IntegrationTests(ITestOutputHelper output)
+        public IntegrationTests(ConnectionSettingsFixture connectionSettingsFixture, ITestOutputHelper output)
         {
             _output = output;
+            ConnectionSettingsFixture = connectionSettingsFixture;
         }
 
-        [Fact(Skip ="need rabbitmq")]
-        public async Task ReceiveMessage_ServiceCall_Success()
+        [Fact]
+        public void ClassicQueueWithTls_ReceiveMessage_Success()
         {
-            IWebHost host = ServiceHelper.CreateWebHostBuilder<Startup>(_output).Build();
+            IWebHost host = ServiceHelper.CreateWebHostBuilder<ClassicQueueWithTLSStartup>(_output).Build();
             using (host)
             {
-                host.Start();
-                await Task.Delay(3000);
-                MessageQueueHelper.SendMessageInQueue();
+                // When the host is started, queues are created if they do not already exist.
+                // By waiting for the host to initialize completely, we avoid the race condition
+                // of trying to access a queue that has not finished creation.
+                Task.WaitAll(Task.Run(() => host.Start()));
+
```

Comment:
I don't understand what you are doing here. host.Start() doesn't return until the host has completely started. You can [see the code here](https://github.com/dotnet/aspnetcore/blob/b10b40ae644995533e59f40f89893c873b3afa42/src/Hosting/Hosting/src/Internal/WebHost.cs#L130-L133) where it calls StartAsync() and waits for the Task to be completed. By using Task.Run, you are starting a Task running on another thread which calls host.Start(), which calls host.StartAsync(), then blocks the thread until that Task has completed, which then completes the thread from Task.Run(). Task.WaitAll() blocks until the passed in Task has completed. If things aren't completely set up by the time that host.Start() completes normally, then there's some work being completed after Start() returns, and all the rest of the code is just adding in a small bit of delay and things are accidentally working. Can you add a step to enumerate all the queues on the server in MessageQueueHelper.SendMessageToQueue to ensure things are setup with a retry with a delay.

Reply:
I think you're right, the extra code was adding enough delay to ensure things worked. Originally, these tests had a thread delay to ensure setup completed and wrapping host.Start() in a task got things working after removing the delay. I'll add the additional step w/ retry to ensure determinism in our tests.

Reply:
Tests seem to run just fine after replacing the Task wrapper with just `host.Start()`. The RabbitMQ client does not have an api to enumerate queues so I'm leaning towards leaving things as they are and add retry logic within `MessageQueueHelper.SendMessageToQueue` if we see failure messages in the future.